### ISSUE TITLE
実trace連動のライブ・ネットワーク模式図を追加する

### ISSUE THREAD

**Issue by hayatok** [hayatok/100challenge#42]
## Goal
MLPとCNNに実trace連動のニューラルネット模式図を追加し、データと演算がどの順で流れているか直感的に理解できるようにする。

## Context
- 対象: `2026-09-02-3/`「数字がわかるまで」
- MLPには隠れ層と出力層の詳細図があるが、入力から回答までの全体経路が弱い
- CNNには特徴マップの詳細表示があるが、ネットワーク全体の現在位置が見えない

## Constraints
- 模式図は装飾ではなく、保存済み実traceと再生phaseに連動する
- MLPとCNNそれぞれの構造を独立して表現する
- 既存の詳細可視化を置き換えず、全体地図から詳細へつなぐ
- 784入力や全接続を常時展開して線だらけにしない
- layer選択をキーボードでも操作可能にする
- 既存の未コミット変更を含めない

## Done when
- [x] MLPで入力、Flatten、隠れ層、Logits、Softmax、回答の流れが見える
- [x] CNNで入力、Conv、Pool、Dense、Softmax、回答の流れが見える
- [x] 現在・通過済み・未到達のlayerが文言と形で区別できる
- [x] 実trace由来のshape、活性値、予測値を模式図へ表示する
- [x] layer選択で対応する再生phaseと詳細表示へ移動できる
- [x] reduced motionで意味を失わない
- [x] test、lint、型チェック、build、4画面幅のブラウザ検証が成功する
